### PR TITLE
bemoji: init at 0.3.0

### PR DESCRIPTION
**Pull request by Laurent2916**
## Description of changes

https://github.com/marty-oehme/bemoji/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/bemoji/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    rev = version;
+    hash = "sha256-XXNrUaS06UHF3cVfIfWjGF1sdPE709W2tFhfwTitzNs=";
+  };
+
+  strictDeps = true;
+  dontBuild = true;
+
+  installPhase = ''
+    chmod +x bemoji
+    install -D bemoji $out/bin/bemoji
```
Comment (nrabulinski) [inline pkgs/applications/misc/bemoji/default.nix:22]: Install sets the executable bit by default, no need for `chmod`. Even if it didn't, if you're using `install` just use the `-m` flag to set the required bits.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -40537,6 +40537,8 @@ with pkgs;
 
   bemenu = callPackage ../applications/misc/bemenu { };
 
+  bemoji = callPackage ../applications/misc/bemoji { };
```
Comment (nrabulinski) [inline pkgs/top-level/all-packages.nix:40540]: New top-level package with no special requirements. Should just be in `by-name`

**Review by nrabulinski — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/bemoji/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/applications/misc/bemoji/default.nix:6]:
```suggestion
stdenvNoCC.mkDerivation rec {
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/bemoji/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  pname = "bemoji";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "marty-oehme";
+    repo = pname;
```
Comment (eclairevoyant) [inline pkgs/applications/misc/bemoji/default.nix:12]:
```suggestion
    repo = "bemoji";
```

pname and repo are semantically different

File: pkgs/applications/misc/bemoji/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  installPhase = ''
+    chmod +x bemoji
+    install -D bemoji $out/bin/bemoji
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/marty-oehme/bemoji/";
+    license = licenses.mit;
+    description = "Emoji picker with support for bemenu/wofi/rofi/dmenu and wayland/X11";
+    maintainers = with maintainers; [ laurent-f1z1 ];
```
Comment (eclairevoyant) [inline pkgs/applications/misc/bemoji/default.nix:29]:
```suggestion
    mainProgram = "bemoji";
    maintainers = with maintainers; [ laurent-f1z1 ];
```

`meta.mainProgram` should be set in the case of a package with some main binary (see #246386)

File: pkgs/applications/misc/bemoji/default.nix (hunk 1 above)
Comment (eclairevoyant) [reply pkgs/applications/misc/bemoji/default.nix:22]:
```suggestion
    install -Dm555 bemoji -t $out/bin
```
and the `chmod` can be removed

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/bemoji/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    owner = "marty-oehme";
+    repo = pname;
+    rev = version;
+    hash = "sha256-XXNrUaS06UHF3cVfIfWjGF1sdPE709W2tFhfwTitzNs=";
+  };
+
+  strictDeps = true;
+  dontBuild = true;
+
+  installPhase = ''
```
Comment (eclairevoyant) [inline pkgs/applications/misc/bemoji/default.nix:20]:
```suggestion
  postInstall = ''
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Review by eclairevoyant — APPROVED** [review]

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 261852` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bemoji</li>
  </ul>
</details>

**Pull request merged**